packages/view-types/heatmap/src/HeatmapSubscriber.js

```diff
@@ -176,8 +183,17 @@ export function HeatmapSubscriber(props) {
   }, []);
 
   const onHeatmapClick = () => {
-    setGeneSelection([geneHighlight]);
-    setCellColorEncoding('geneSelection');
+    if (hoveredColorEncoding === 'geneSelection') {
+      setGeneSelection([geneHighlight]);
+      setCellColorEncoding('geneSelection');
+    } else if (hoveredColorEncoding === 'cellSelection') {
+      const selectionFullPath = getObsMembership(cellHighlight);
+      const selectionToHighlight = findLongestCommonPath(selectionFullPath, cellSetSelection);
+      if (selectionToHighlight) {
+        setCellSetSelection([selectionToHighlight]);
+        setCellColorEncoding('cellSelection');
+      }
+    }
```

Comment:
```suggestion
    if (hoveredColorEncoding === 'geneSelection' && geneHighlight) {
      setGeneSelection([geneHighlight]);
      setCellColorEncoding('geneSelection');
    } else if (hoveredColorEncoding === 'cellSelection' && cellSetSelection) {
      const selectionFullPath = getObsMembership(cellHighlight);
      if (selectionFullPath?.length > 0) {
        const selectionToHighlight = findLongestCommonPath(selectionFullPath, cellSetSelection);
        if (selectionToHighlight) {
          setCellSetSelection([selectionToHighlight]);
          setCellColorEncoding('cellSelection');
        }
      }
    }
```

Prefer extra checks to be on the safe side in edge cases where cell sets are not available, for example.